src/Microsoft.OData.Edm/Schema/CoreModel/EdmCoreModelComplexType.cs

```diff
@@ -0,0 +1,86 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmCoreModelComplexType.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// The built-in Edm.ComplexType abstrat type in the core model.
+    /// </summary>
+    internal sealed class EdmCoreModelComplexType : EdmType, IEdmComplexType, IEdmCoreModelElement
+    {
+        /// <summary>
+        /// The core Edm.ComplexType singleton.
+        /// </summary>
+        public static readonly EdmCoreModelComplexType Instance = new EdmCoreModelComplexType();
+
+        /// <summary>
+        /// Gets the kind of this type.
+        /// </summary>
+        public override EdmTypeKind TypeKind => EdmTypeKind.Complex;
+
+        /// <summary>
+        /// Gets the schema element kind of this type.
+        /// </summary>
+        public EdmSchemaElementKind SchemaElementKind => EdmSchemaElementKind.TypeDefinition;
+
+        /// <summary>
+        /// Gets the name of this type.
+        /// </summary>
+        public string Name => EdmCoreModelConstants.ComplexType;
+
+        /// <summary>
+        /// Gets the namespace of this type.
+        /// </summary>
+        public string Namespace => EdmCoreModelConstants.EdmNamespace;
+
+        /// <summary>
+        /// Gets a value indicating whether this type is abstract.
+        /// The Edm.ComplexType is always an abstract type.
+        /// </summary>
+        public bool IsAbstract => true;
+
+        /// <summary>
+        /// Gets a value indicating whether this type is open.
+        /// The Edm.ComplexType is always non-open type.
+        /// </summary>
+        public bool IsOpen => false;

```

Comment:
Why do we assume Edm.ComplexType is always non-open?

Reply:
What I am thinking is that to mark the Edm.ComplexType as open type doesn't make any sense, because I think the Edm.ComplexType doesn't have any property. But, I can't find the info from the Spec to prove myself. So, what do you think? @mikepizzo

Reply:
Synced at Standup. It's fine to make it as false for abstract type.

---
In reply to: [164582848](https://github.com/OData/odata.net/pull/1051#discussion_r164582848) [](ancestors = 164582848)